pkg/process/util/network_filter.go

```diff
@@ -0,0 +1,102 @@
+package util
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ConnectionFilter holds a user-defined blacklisted IP/CIDR, and ports
+type ConnectionFilter struct {
+	IP       *net.IPNet
+	Ports    map[uint16]struct{}
+	AllPorts bool
+}
+
+// ParseConnectionFilters takes the user defined blacklist and returns a slice of ConnectionFilters
+func ParseConnectionFilters(filters map[string][]string) (blacklist []*ConnectionFilter) {
+	for ip, ports := range filters {
+		filter := &ConnectionFilter{Ports: map[uint16]struct{}{}}
+		var subnet *net.IPNet
+		var err error
+
+		// retrieve valid IPs
+		if strings.ContainsRune(ip, '*') {
+			subnet = nil // use for wildcard
+		} else if strings.ContainsRune(ip, '/') {
+			_, subnet, err = net.ParseCIDR(ip)
+		} else if strings.ContainsRune(ip, '.') {
+			_, subnet, err = net.ParseCIDR(ip + "/32") // if given ipv4, prefix length of 32
+		} else if strings.Contains(ip, "::") {
+			_, subnet, err = net.ParseCIDR(ip + "/64") // if given ipv6, prefix length of 64
+		} else {
+			log.Debugf("Invalid IP/CIDR/* defined for connection filter: %s", err)
+			continue
+		}
+
+		if err != nil {
+			log.Debugf("Could not parse given IPs: %s", err)
+			continue
+		}
+		filter.IP = subnet
+
+		validFilter := true
+		for _, v := range ports {
+			if v == "*" {
+				// This means that IP + port are both *, which effectively blacklists all conns, which is invalid.
+				if subnet == nil {
+					log.Debugf("Invalid filter with IP/CIDR as * and port as *: %s", err)
```

Comment:
same comment as above, should b Error and explain that the rule won't be respected